docs/generate-features.md

```diff
@@ -0,0 +1,46 @@
+#### generate-features
+---
+Scan the class files of your application and create a file containing the Liberty features that it requires.
+
+This goal is available as a tech preview in the 3.5.2-SNAPSHOT. Please provide feedback by opening an issue at https://github.com/OpenLiberty/ci.maven.
+
+This feature is best accessed through dev mode. When you start up `liberty:dev` it will compile your application and scan the files to verify that all the Liberty features that you need are part of your configuration. Then as you work, dev mode will continue to monitor the project to confirm the Liberty features are up to date. If you implement a new interface in Java the scanner will determine if that API is one of Liberty's, update the configuration and install the feature. If you remove a feature from `server.xml`, dev mode will determine if it is actually necessary and if so add it to the configuration file described below. For this snapshot you need to add the Sonatype repository to `pom.xml` (shown below) but in the future all the dependencies will be in Maven Central.
+
+Lifecycle
+
+This goal is not part of the Maven lifecycle so to use it in your build you will need to understand its dependencies. Since it will scan the class files of your application it must be run after you have run the compile goal. The list of features that it generates will be used by the `liberty:create` and the `liberty:install-feature` goals so run this goal first.
+
+If the goal detects features which are not already in the Liberty configuration it will create the file `configDropins/overrides/generated-features.xml` in the `src/main/liberty/config` section of your project. The goal will add the necessary features which are not already in the configuration. If the `generated-features.xml` file has been created in the past but all the Liberty features used by the application are already specified elsewhere in the configuration this file will be retained and will contain a comment explaining this situation.
```

Comment:
Consider rephrasing to:

If this goal detects Liberty features used in your project but not already present in your Liberty configuration it will create a new file `configDropins/overrides/generated-features.xml` in the `src/main/liberty/config` directory of your project. The `generated-features.xml` file will contain a list of features required for your project. If the `generated-features.xml` file has been created in the past and no additional features have been detected, this file will be retained and will contain a comment indicating that there are no additional features generated.

Reply:
I like this.